engine/src/execution_graph/logic_controllers/BatchProcessing.cpp

```diff
@@ -123,6 +124,11 @@ kstatus TableScan::run() {
     std::vector<int> projections(schema.get_num_columns());
     std::iota(projections.begin(), projections.end(), 0);
 
+    if (provider->is_sql()) {
+      auto sql_provider = std::dynamic_pointer_cast<ral::io::abstractsql_data_provider>(provider);
```

Comment:
This should happen at construction since we can determine that the moment the provider is created so it is not needed to do this casting or set these column indices here. that should happen during construction

Reply:
also and more importantly, the kernels shoudl have no idea of what kind of provider they have or dont have. They should only use the functions that are common to the provider base class. That is part of the intention of these classes. Yoiu shouldnt need to add anything to them

Reply:
I've removed is_sql from the data_provider interface, done